src/App.tsx

```diff
@@ -122,116 +129,87 @@ function Main() {
-        <span className="mx-2">-</span>
-        <strong>Valid:</strong>
-        <span>{result.status === IdStatus.Valid ? "✅" : "❌"}</span>
-        <br />
-        {result.status === IdStatus.Incomplete && (
-          <em style={{ color: "grey" }}>
-            {"example: "}
-            <span
-              className="fillin"
-              data-example={result.example}
-              onClick={(event) => {
-                handleGrabEg(event);
-              }}
-            >
-              {result.example}
-            </span>
-          </em>
-        )}
-      </div>
-    );
-  }
-
   return (
     <div className="page-center">
       <div className="resolve-panel">
-        {/* Intro icons and text */}
         <h1 className="text-large">🏷️ ➜ 📦</h1>
         <p>
           The FC4EOSC Metaresolver resolves individual handles from various
           providers
         </p>
-        {/* Pid input box */}
         <input
           type="text"
           className="form-control"
           style={{ maxWidth: "600px" }}
           id="input-pid"
           value={pid}
-          onChange={(event) => {
-            handleChange(event);
-          }}
+          onChange={handleChange}
           autoComplete="off"
         />
-        <div className="info">{tooltip}</div>
+        <div className="info">
+          {results.length === 0 ? (
+            <div style={{ color: "grey" }}>
+              <strong>Please enter a valid Pid</strong>
+              <br />
+              <em>
+                supported pids: ark, arXiv, swh, doi, urn:nbn{" "}
+                <Link to="/supported-pids"> and more...</Link>
+              </em>
+            </div>
+          ) : (
+            results.map((result, index) => (
+              <div key={index}>
+                <strong>Format: </strong>
+                <span>{result.type + " "}</span>
+                <span className="mx-2">-</span>
+                <strong>Valid:</strong>
+                <span>{result.status === IdStatus.Valid ? "✅" : "❌"}</span>
+                <br />
+                <div className="m-2" style={{ color: "grey" }}>
+                  {"example: "}
+                  <span
+                    className="fillin"
+                    data-example={result.example}
+                    onClick={handleGrabEg}
+                  >
+                    {result.example}
+                  </span>
+                </div>
+                <div>
+                  {sortResolutionModes(result.resolution_modes).map(
+                    (mode, idx) => (
+                      <a
+                        key={idx}
+                        className={`resolve-btn btn btn-lg btn-primary mr-2 ${
+                          result.status === IdStatus.Valid ? "" : "disabled"
+                        }`}
+                        href={generateResolveURL(
+                          mode.mode as ResolutionModes,
+                          result.example,
```

Comment:
Resolve url should include user's input value (not API's result.example)

Reply:
Fixed